debug/gdbserver.py

```diff
@@ -567,6 +567,13 @@ def test(self):
         assertEqual(123, self.gdb.p("$x1"))
         assertEqual(123, self.gdb.p("$csr832"))
 
+class NonExistentCsr(RegsTest):
+    def test(self):
+        #Print old version of $misa at 0xF10 (moved to 0x301)
+        self.gdb.p("$csr3856")
```

Comment:
Your test returned an exception. There should be more output that details exactly what happened, but I suspect you hit this code in p():
```
    def p(self, obj):
        output = self.command("p/x %s" % obj)
        m = re.search("Cannot access memory at address (0x[0-9a-f]+)", output)
        if m:
            raise CannotAccess(int(m.group(1), 0))
        value = int(output.split('=')[-1].strip(), 0)
        return value
```

So your tests needs to:
```
try:
   self.gdb.p(...)
   assert False, "access should have failed"
except CannotAccess:
   # expected
   pass
```

just like MemTestReadInvalid